File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/main.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int inc_length = 1, dec_length = 1, max_length = 1;

        for (size_t i = 0; i < nums.size() - 1; i++) {
            if (nums[i + 1] > nums[i]) {
                inc_length++;
                dec_length = 1;

            }else if(nums[i + 1] < nums[i]) {
                dec_length++;
                inc_length = 1;
            }else {
                inc_length = dec_length = 1;
            }

            max_length = std::max(max_length, std::max(inc_length, dec_length));
        }

        return max_length;
    }
};

File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/main.go
func longestMonotonicSubarray(nums []int) int {
    incLength, decLength, maxLength := 1, 1, 1

    for i := 0; i < len(nums) - 1; i++ {
        if nums[i + 1] > nums[i] {
            incLength++
            decLength = 1
        }else if nums[i + 1] < nums[i] {
            decLength++
            incLength = 1
        }else {
            incLength = 1
            decLength = 1
        }

        maxLength = max(maxLength, max(incLength, decLength))
    }

    return maxLength
}

func max(a, b int) int {
    if a < b {
        return b
    }

    return a
}
